### ISSUE TITLE
tesseract::TessBaseAPI::ProcessPages cannot be stopped on demand

### ISSUE THREAD

**Issue by flaviu22** [tesseract-ocr/tesseract#3849]
### Environment

* **Tesseract Version**: Tesseract 4.1.1
* **Platform**: Win10 64bit, VS2017, MFC C++ application

### Current Behavior:
**tesseract::TessBaseAPI::ProcessPages** cannot be stopped on demand. I didn't discover any way to stop **tesseract::TessBaseAPI::ProcessPages** when I want, some timeout could be setup, but this not help
### Expected Behavior:
**tesseract::TessBaseAPI::ProcessPages** should have a solution to be stopped on demand
### Suggested Fix:
**tesseract::TessBaseAPI::ProcessPages** should taking account **ETEXT_DESC** somehow. More details could be found here: https://stackoverflow.com/questions/72719440/stop-tesseracttessbaseapiprocesspages-on-demand

**Comment by amitdo** [comment]
The timeout does not work the way you think it does.

The whole OCR process is done in a few phases.
* Binarization
* Image finding
* Horizontal and vertical line finding
* Layout Analysis
* Text recognition

The time is checked before each phase starts, but there is  no way to stop tesseract in the **middle** of a phase (except manually killing the whole process).

**Comment by flaviu22** [comment]
So, in this case there no way to stop _tesseract::TessBaseAPI::ProcessPages_ gracefully when I want, so, you can close the issue. If you have some hint here, please tell me. Thanks.

**Comment by amitdo** [comment]
This timeout feature can be improved by adding a few sub-phases.

I marked this issue as 'feature request', but it will probably have low priority.

**Comment by flaviu22** [comment]
> 

Yes, that's will be perfect, I wonder if I can imply myself, to solve this as quickly as possible.
